### DISCUSSION TITLE
Break Auro Icons into separate repos; auro icons and auro logos

### DISCUSSION THREAD

**Discussion by blackfalcon** [AlaskaAirlines/WC-Generator#520] [Announcements]
## The problem

re: https://github.com/AlaskaAirlines/Icons/issues/106

## The recommendation 

tl;dr - break the current auro icons into two repos. Auro icons will be released with a MAJOR version change and renamed to `@aurodesignsystem/icons`. The new repo will be `@aurodesignsystem/logos`. 

### Team impact

Heavy consumers of the Auro icons repo use the assets within the node package via a proxy. 

1. **Flight search** - subscribes to icons and logos via the team's personal CDN solution. Additionally, a custom web component called `airline-helper` is used to display logos/tails and that web component also subscribes to the same CDN solution. 
    * The CDN is populated with assets from the node package via a custom pipeline. It is assumed that the pipeline could be updated to pull from two packages versus the current one and there would be no need for long-tail UI code changes or any interruption of service.  
1. **Direct CDN use** - if a consumer is referencing a icon or logo via a URL that links directly back to an asset via the CDN, there will be no impact at all. Even if the consumer is using `@latest`, there will be no changes to the `@alaskaairux` namespaced package and the URLs will not stop working. It will be up to the consumer to edit their reference as needed. 
1. **Content** - content producers make use of the `auro-icon` custom element. This element allows for the use of either icons, tails or logos. To support this change, `auro-icon` would need to be updated to pull assets from one CDN location versus the other. See the following code ref to be updated. In this use case the `auro-icon` element takes on the work of the breaking change in Auro Icons and the content team would only need to upgrade `auro-icon` to support. 
    * Prior to the breaking changes, we could release an update to `auro-icon` that hard-codes the version of Icons used versus `@latest` to ensure transition stability.  

<img width="868" alt="Screen Shot 2024-02-26 at 11 58 26 AM" src="https://github.com/AlaskaAirlines/WC-Generator/assets/181089/0d329366-6914-4de7-a81b-23ea9f97ff7e">

Any consumer of the node repackage directly in their code will need to update their code when upgrading to the new MAJOR release. That is considered standard practice with the clear communication that this is a BREAKING CHANGE. 

**Optional new feature** - with the new auro logos repo, we could also set up intentional separation of logos from tail designs. E.g. `./dist/tails` and `./dist/partner-logos`. Also, what about logo categories? 

## The expectation 

The expected outcome of this work is to ensure that both auro icons and the newly proposed auro logos are individual repos with their own paths of growth. Removing this complication will reduce the efforts needed to maintain either repo.
